inhouse/command_handlers/soloqueue_leaderboard.py

```diff
@@ -118,18 +122,32 @@ async def make(self, emojiList):
             await self.channel.send(embed=msg)
         await self.channel.send(view=JoinButtons(self.db_handler))
 
-    def num_ranked_past_week(self, name: str):
+    def num_ranked_past_week(self, puuid):
         try:
-            response = inhouse.global_objects.watcher.summoner.by_name(self.my_region,summoner_name=name) 
-            puuid = response['puuid']
             # subtract number of seconds in a week to get a week ago epoch time
             week_ago = int(time.time() - inhouse.constants.seconds_in_week)
             #queue type 420 is "solo queue" according to riot API documentation (lol)
             response = inhouse.global_objects.watcher.match.matchlist_by_puuid(self.my_region,puuid=puuid,start_time=week_ago,queue=420,count=100)
+            time.sleep(0.05)
             return len(response)
         except Exception as e:
                 print(e)
 
+    # rate limits are 20 every second and 100 every 2 min, since we are calling API twice its 10 every second and 100 every 2 minutes
+    @limits(calls=10, period=1)
+    @limits(calls=100, period=inhouse.constants.seconds_in_two_min)
+    def api_calls(self, sum_id, puuid):
```

Comment:
I believe there is also a `@sleep_and_retry` decorator we can use to avoid the exception handle case